pkg/agent/controller/networkpolicy/allocator_test.go

```diff
@@ -143,3 +166,54 @@ func TestRelease(t *testing.T) {
 		})
 	}
 }
+
+func TestWorker(t *testing.T) {
+	rule := &types.PolicyRule{
+		Direction: v1beta2.DirectionIn,
+		From:      []types.Address{},
+		To:        ofPortsToOFAddresses(sets.NewInt32(1)),
+		Service:   nil,
+	}
+	tests := []struct {
+		name        string
+		args        []uint32
+		rule        *types.PolicyRule
+		expectedID  uint32
+		expectedErr error
+	}{
+		{
+			"delete-rule-from-async-rule-cache",
+			nil,
+			rule,
+			1,
+			nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			a := newIDAllocator(tt.args...)
+			actualErr := a.allocateForRule(tt.rule)
+			if actualErr != tt.expectedErr {
+				t.Fatalf("Got error %v, expected %v", actualErr, tt.expectedErr)
+			}
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			go wait.Until(a.worker, time.Millisecond, stopCh)
+
+			a.asyncRuleCache.deleteQueue.AddAfter(tt.rule.FlowID, testAsyncDeleteInterval)
```

Comment:
it's a bit of a nit, but I feel like the test should call `forgetRule` instead of accessing the queue directly, and that the deletion interval should be a `var` (instead of a `const`) so it can be set to a specific value when testing. This is what we do in other tests, e.g. https://github.com/vmware-tanzu/antrea/blob/master/pkg/apiserver/certificate/certificate.go#L49

Reply:
Makes sense. Done.